src/backend/base/langflow/services/database/service.py

```diff
@@ -93,7 +94,13 @@ def _create_engine(self) -> AsyncEngine:
                 "pool_size": self.settings_service.settings.pool_size,
                 "max_overflow": self.settings_service.settings.max_overflow,
             }
-            database_url = "postgresql+psycopg://" if url_components[0].startswith("postgresql") else url_components[0]
+            database_url = (
+                "postgresql+psycopg://"
+                if url_components[0].startswith("postgresql")
+                else f"{self.database_driver_async}://"
+                if self.database_driver_async
```

Comment:
If provided, `database_driver_async` should probably be the first choice.

Reply:
`database_driver_async` is still not the first choice. It should take priority over `postgresql+psycopg://`